app/assets/javascripts/classes/MetricsBrowseList.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ko = require('knockout');
+import ns = require('./NaturalSort');
+import ServiceData = require('./ServiceData');
+import ServiceNodeVM = require("./ServiceNodeVM");
+import GraphSpec = require("./GraphSpec");
+import FolderNodeVM = require("./FolderNodeVM");
+import MetricNodeVM = require("./MetricNodeVM");
+import StatisticNodeVM = require("./StatisticNodeVM");
+
+class MetricsBrowseList {
+    serviceNodes: KnockoutObservableArray<ServiceNodeVM> = ko.observableArray<ServiceNodeVM>();
+
+    private idify(value: string): string {
+        if (value == undefined) {
+            return undefined;
+        }
+        value = value.replace(/ /g, "_").toLowerCase();
+        return value.replace(/\//g, "_");
+    };
+
+    private getPathParts(path: string): string[] {
```

Comment:
not sure what our js style ordering is here, but the methods seem to be in inconsistent order based on scope here.